File: Binary Search.py
def binarySearch(arr, lower, upper, x):
  if upper >= 1:
    mid = (lower + upper)//2
    
    if arr[mid] == x:
      return mid
    elif arr[mid] > x:
      return binarySearch(arr, lower, mid-1, x)
    else:
      return binarySearch(arr, mid+1, upper, x)
    
    return -1
  
arr = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20]
value = 10
result = binarySearch(arr, 0, len(arr)-1, value)

if result != -1:
  print(result)
else:
  print("Number Not Exist!")

File: Algorithms/src/main/java/searching/BinarySearch.java
package searching;

/**
 * Created by Imran on 2/2/2020.
 */
public class BinarySearch {

    int binarySearch(int arr[], int low, int high, int x)
    {
        if (high >= low) {
            int mid = low + (high - low) / 2;
            if (arr[mid] == x)
                return mid;
            if (arr[mid] > x)
                return binarySearch(arr, low, mid - 1, x);
            return binarySearch(arr, mid + 1, high, x);
        }
        return -1;
    }

    public static void main(String args[])
    {
        BinarySearch ob = new BinarySearch();
        int arr[] = { 2, 3, 4, 10, 40 };
        int n = arr.length;
        int x = 10;
        int result = ob.binarySearch(arr, 0, n - 1, x);
        if (result == -1)
            System.out.println("Element not present");
        else
            System.out.println("Element found at index " + result);
    }
}
